SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex

```diff
@@ -3284,6 +3284,11 @@ \section{Using Standard Macros}
 \item \code{ITK\_DISALLOW\_COPY\_AND\_ASSIGN(TypeName)}: Disallow copying by
 declaring copy constructor and assignment operator deleted. This must be
 declared in the \textbf{public} section.
+\item \code{ITK\_DEFAULT\_COPY\_AND\_MOVE(TypeName)}: Enables copying and moving
+by explicitly defaulting the copy constructor, copy assignment operator, move
+constructor, and move assignment operator of a class. Especially useful for
+classes that have a user-defined destructor. Intended to be placed in the
+\textbf{public} section of a class.
```

Comment:
> Thanks for doing this Niels. Are all classes expected to have this?

You're welcome Jon. In my opinion, this macro ITK_DEFAULT_COPY_AND_MOVE is especially useful for classes that have a user-defined destructor. You know, it's meant to address those warnings that you reported at https://github.com/InsightSoftwareConsortium/ITK/issues/4645 More details can be found at the doxygen/API documentation: https://github.com/InsightSoftwareConsortium/ITK/blob/eb490ff5520a4ed42fa607c5e72b2e58518407d2/Modules/Core/Common/include/itkMacro.h#L377-L383

When all "special member functions" of class (constructors, assignment operators, destructor) can be generated automatically by the compiler, I think the "Rule of Zero" is preferable to using this macro.

Reply:
OK, I see now that it is mentioned in the explanation of the macro. Thanks Niels.